heat/utils/data/matrixgallery.py

```diff
@@ -1,153 +1,186 @@
-                type(singular_values)
-            )
-        )
-    if not singular_values.ndim == 1:
-        raise RuntimeError(
-            "Argument singular_values needs to be a 1D array, but dimension is {}.".format(
-                singular_values.ndim
-            )
-        )
-    if singular_values.shape[0] > min(m, n):
-        raise RuntimeError(
-            f"Number of given singular values must not exceed matrix dimensions. Got {singular_values.shape[0]} singular values for matrix size ({m}, {n})."
-        )
-
-    r = singular_values.shape[0]
-    U = random_orthogonal(m, r, split=split, device=device, comm=comm, dtype=dtype)
-    V = random_orthogonal(n, r, split=split, device=device, comm=comm, dtype=dtype)
-
-    A = matmul(U, matmul(diag(singular_values), V.T))
-
-    return A.resplit_(split), (U, singular_values, V)
-
-
-def random_known_rank(
-    m: int,
-    n: int,
-    r: int,
-    quantile_function: Callable = lambda x: -log(x),
-    split: Union[None, int] = None,
-    device: Union[None, str, Device] = None,
-    comm: Union[None, Communication] = None,
-    dtype: Type[datatype] = core.float32,
-) -> Tuple[DNDarray, Tuple[DNDarray]]:
-    """
-    Creates a random m x n matrix with rank r.
-    This routine uses :func:`random_known_singularvalues` with r singular values randomly chosen
-    w.r.t. the distribution with quantile function given by the input quantile_function. Default yields exponential distibution with parameter lambda=1.
-    Unlike in :func:`random_known_singularvalues`, here the singular values of the output are sorted in descending order.
-    """
-    if r > min(m, n):
-        raise RuntimeError("rank must not exceed matrix dimensions.")
-
-    singular_values = rand(r, dtype=dtype, comm=comm, device=device)
-    singular_values = sort(quantile_function(singular_values), descending=True)[0]
-
-    return random_known_singularvalues(
-        m, n, singular_values, split=split, device=device, comm=comm, dtype=dtype
-    )
+"""
+Generate matrices for specific tests and functions
+"""
+
+from heat import core
+from ...core.dndarray import DNDarray
+from ...core.communication import Communication
+from ...core.devices import Device
+from ...core.types import datatype
+from ...core.random import randn, rand
+from ...core.linalg import qr, matmul
+from ...core.manipulations import diag, sort
+from ...core.exponential import log
+from typing import Type, Union, Tuple, Callable
+
+__all__ = ["hermitian", "parter", "random_known_singularvalues", "random_known_rank"]
+
+
+def hermitian(
+    n: int,
+    split: Union[None, int] = None,
+    device: Union[None, str, Device] = None,
+    comm: Union[None, Communication] = None,
+    dtype: Type[datatype] = core.complex64,
+) -> DNDarray:
+    """
+    Generates a Hermitian matrix of size `(n,n)`. A Hermitian matrix is a  complex square matrix that is equal to its conjugate transpose.
```

Comment:
With a small modification (devide by 1/size) we could add here that the matrices are sampled according to the "Marchenko-Pastur distribution" (https://en.wikipedia.org/wiki/Marchenko%E2%80%93Pastur_distribution)